lib/ansible/module_utils/facts/system/iscsi.py

```diff
@@ -0,0 +1,52 @@
+# iSCSI initiator related facts collection for Ansible.
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import subprocess
+import sys
+
+from ansible.module_utils.facts.utils import get_file_content
+from ansible.module_utils.facts.collector import BaseFactCollector
+
+
+class IscsiInitiatorFactCollector(BaseFactCollector):
+    name = 'iscsi'
+    _fact_ids = set()
+
+    def collect(self, module=None, collected_facts=None):
+        iscsi_facts = {}
+        iscsi_facts['iscsi_iqn'] = {}
+        if sys.platform.startswith('linux') or sys.platform.startswith('sunos'):
```

Comment:
Could you provide some sample versions of the text from these commands/files?

( /etc/iscsi/initiatorname.iscsi, `lsattr -E -l iscsi0` output, etc)